test/redirect/server.test.js

```diff
@@ -0,0 +1,60 @@
+require('../../src/redirect/config');
+const supertest = require('supertest');
+const { server } = require('../../src/redirect/server');
+
+const EXAMPLE_REQUEST_BODY = require('./fixtures/exampleRequestBody.json');
+const INVALID_ID = 'INVALID_ID';
+describe('test /qr endpoint', () => {
+  test('POST request to /qr returns 200 status with claim url', async () => {
+    await supertest(server)
+      .post('/qr')
+      .send(EXAMPLE_REQUEST_BODY)
+      .expect(200)
+      .then(response => {
+        expect(response.body).toHaveProperty('url');
+      });
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+});
+
+describe('test /qr/[qr id]/claim endpoint', () => {
+  test('POST request to /qr/[qr id]/claim returns 403 status for invalid/previously claimed QR', async () => {
+    await supertest(server).post(`/qr/${INVALID_ID}/claim`).expect(403);
+  });
+
+  test('Valid POST request to /qr/[qr id]/claim returns 301 status', async () => {
+    const qrResponse = await supertest(server).post('/qr').send(EXAMPLE_REQUEST_BODY).expect(200);
+    await supertest(server)
+      .post(qrResponse.body.url.replace('http://localhost:3000', ''))
+      .expect(301)
+      .then(response => {
+        expect(response.header.location).toBeDefined();
+      });
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+});
+
+describe('test /qr/[qr id]/claimed/[claim id]/files/[file id] endpoint', () => {
+  test('GET request to valid file location returns file content from /qr request', async () => {
+    const qrResponse = await supertest(server).post('/qr').send(EXAMPLE_REQUEST_BODY).expect(200);
```

Comment:
I don't know if we need the intermediate `expect()` assertions here and on line 48 since those were already tested and this specific test case is intended to test just the get request below. Now that I say that, if you kept them here, you could probably get rid of the tests starting on line 28 and on line 8, and just move the last `expect()` statements from those ones to this test and test everything sequentially.

Open to whichever way you want to do it, but I don't think we want to be "expect"ing the same thing in multiple different tests.